### DISCUSSION TITLE
Can we exposed the mutable referance of CommandEncoder in RenderPass?

### DISCUSSION THREAD

**Discussion by Ahuang0107** [gfx-rs/wgpu#6461] [Ideas]
Below is an example of what I'm trying to do.
In some cases, I hope to call command_encoder's copy_texture_to_texture once between processing instance A and instance B after creating render_pass. But it's currently not possible because RenderPass holds a mutable reference to CommandEncoder. If we can exposed RenderPass.parent, the following code can run.
<img width="853" alt="Screenshot 2024-10-24 194219" src="https://github.com/user-attachments/assets/44ffd263-14bc-4428-8778-680c89fbc043">
(ps: of course, in the following example, I can modify it to create a separate pass when processing each instance, but I actually encountered this problem when using bevy. it will be much simple to do some adjust in wgpu)

**Comment by grovesNL** [comment]
Hi! :wave: Usually the pattern is that you perform any copies or anything that needs exclusive access to the render pass before you start the render pass. So you could have a separate loop before `let mut render_pass` that performs all of the copies.

**Reply by Ahuang0107** [reply]
thanks for reply, you are right, i try to build wgpu my self and exposed &mut CommandEncoder, it does not run as a i expected. i need to do copies before render pass.
